server/master/src/main/java/org/apache/accumulo/master/tableOps/bulkVer2/LoadFiles.java

```diff
@@ -136,6 +143,52 @@ void start(Path bulkDir, Master master, long tid, boolean setTime) throws Except
       fmtTid = String.format("%016x", tid);
 
       loadMsgs = new MapCounter<>();
+
+      loadQueue = new HashMap<>();
+    }
+
+    private void sendQueued(int threshhold) {
+      if (queuedDataSize > threshhold || threshhold == 0) {
+        loadQueue.forEach((server, tabletFiles) -> {
+
+          if (log.isTraceEnabled()) {
+            log.trace("tid {} asking {} to bulk import {} files for {} tablets", fmtTid, server,
+                tabletFiles.values().stream().mapToInt(Map::size).sum(), tabletFiles.size());
+          }
+
+          TabletClientService.Client client = null;
+          try {
+            client = ThriftUtil.getTServerClient(server, master.getContext(), timeInMillis);
+            client.loadFiles(Tracer.traceInfo(), master.getContext().rpcCreds(), tid,
+                bulkDir.toString(), tabletFiles, setTime);
+          } catch (TException ex) {
+            log.debug("rpc failed server: " + server + ", tid:" + fmtTid + " " + ex.getMessage(),
+                ex);
+          } finally {
+            ThriftUtil.returnClient(client);
+          }
+        });
+
+        loadQueue.clear();
+        queuedDataSize = 0;
+      }
+    }
+
+    private void queueLoad(HostAndPort server, KeyExtent extent,
```

Comment:
This is a confusing name, since we have different meanings for the term "load".  I assume you are "loading the queue" here but perhaps a name like addToQueue may be less confusing.

Reply:
Changed  in f2d62d7